src/useFormState.test.tsx

```diff
@@ -433,6 +433,63 @@ describe("useFormState", () => {
     expect(autoSaves).toEqual(1);
   });
 
+  it("can queue up changes for auto save if a save is already in progress", async () => {
+    const autoSaveStub = jest.fn();
+    type FormValue = Pick<AuthorInput, "id" | "firstName" | "lastName">;
+    const config: ObjectConfig<FormValue> = {
+      id: { type: "value" },
+      firstName: { type: "value" },
+      lastName: { type: "value" },
+    };
+
+    // Given a component using `getObjectState` for lazily creating ObjectStates
+    function TestComponent() {
+      const [apiData, setApiData] = useState<FormValue>({ id: "a:1", firstName: "Brandon", lastName: "Dow" });
+      const state = useFormState({ config, autoSave, init: { input: apiData, map: (v) => v } });
+      async function autoSave(form: ObjectState<FormValue>) {
+        autoSaveStub(form.changedValue);
+        // Pretend to make an API call and update the local state
+        await setApiData((prevState) => ({ ...prevState, ...form.changedValue }));
```

Comment:
Fwiw I'm not sure this `await setApiData` is really adding any async behavior in the test, b/c the flow will be:

- Step one call `setApiData` to get `some result`
- Await `some result`

Well, I take that back, it will mean that `autoSave` has to at least go through 1 cycle of the promise queue, so it will not immediately return.

Something like:

```
setApiData(...);
await Promise.resolved(1);
```

Would have the same effect and look a little bit more kosher I think.

Reply:
Sure, that sounds good.